config.go

```diff
@@ -161,85 +160,123 @@ func (config TGFConfig) String() string {
 	return string(bytes)
 }
 
-func (config *TGFConfig) getAwsSession(duration int64) (*session.Session, error) {
-	if cachedSession != nil {
-		return cachedSession, nil
-	}
-	askedForMfa := false
-	options := awsSession.Options{
-		Profile:           config.tgf.AwsProfile,
-		SharedConfigState: awsSession.SharedConfigEnable,
-		AssumeRoleTokenProvider: func() (string, error) {
-			askedForMfa = true
-			fmt.Fprintf(os.Stderr, "Assume Role MFA token code: ")
-			v, err := terminal.ReadPassword(int(os.Stdin.Fd()))
-			fmt.Fprintln(os.Stderr)
-			return string(v), err
-		},
-	}
-	if duration > 0 {
-		options.AssumeRoleDuration = time.Duration(duration) * time.Second
+var cachedAwsConfig *aws.Config
+
+func (tgfConfig *TGFConfig) getAwsConfig(assumeRoleDuration time.Duration) (*aws.Config, error) {
+	if cachedAwsConfig != nil {
+		log.Debug("Using cached AWS config")
+		return cachedAwsConfig, nil
 	}
 
-	session, err := awsSession.NewSessionWithOptions(options)
+	log.Debugf("Creating new AWS config (assumeRoleDuration=%s)", assumeRoleDuration)
+	_config, err := awsConfig.LoadDefaultConfig(
+		context.TODO(),
+		awsConfig.WithSharedConfigProfile(tgfConfig.tgf.AwsProfile),
+		awsConfig.WithAssumeRoleCredentialOptions(func(o *stscreds.AssumeRoleOptions) {
+			o.TokenProvider = stscreds.StdinTokenProvider
```

Comment:
Yes. `stscreds.StdinTokenProvider` is a function defined by the SDK that reads from stdin to fetch an MFA token.

When this is set and you need an MFA the SDK will call the function.

When it's not set and you need an MFA, you'll get an error from the SDK because it's unable to provide an MFA.

When you don't need an MFA the function is just ignored and never called.